include/envoy/stream_info/stream_info.h

```diff
@@ -64,6 +65,22 @@ enum ResponseFlag {
   LastFlag = StreamIdleTimeout
 };
 
+/**
+ * Constants for the response code details field of StreamInfo.
+ *
+ * These provide details about the stream state such as whether the
+ * response is from the upstream or from envoy (in case of a local reply).
+ * Custom extensions can define additional values provided they are appropriately
+ * scoped to avoid collisions.
+ */
+struct ResponseCodeDetailValues {
+  const std::string RC_SET_BY_UPSTREAM = "response_code_set_by_upstream";
```

Comment:
nit: I wonder if we have to prefix RC here given the structs name?

Reply:
Most constants won't need to start with RC but should be a standalone phrase that describes 

> > > @eziskind are you also going to have subsequent PRs to surface this information out, e.g to make this accessible in access logs?
> > 
> > 
> > Access logs already get a [reference](https://github.com/envoyproxy/envoy/blob/b8bcb11f5fb8c5ae19535444a9de10a207d9cec6/include/envoy/access_log/access_log.h#L89) to the StreamInfo object so they will be able to get the response code details from that.
> 
> Right, I know that the access log already has a reference. I am just wondering if you are also going to add functionality to be able to write this new information the stream info has. So for instance in the access log case are you planning on adding it to the formatter:
> 
> [envoy/source/common/access_log/access_log_formatter.cc](https://github.com/envoyproxy/envoy/blob/de039269f54aa21aa0da21da89a5075aa3db3bb9/source/common/access_log/access_log_formatter.cc#L257)
> 
> Line 257 in [de03926](/envoyproxy/envoy/commit/de039269f54aa21aa0da21da89a5075aa3db3bb9)
> 
>  StreamInfoFormatter::StreamInfoFormatter(const std::string& field_name) { 
> Just wondering how you are planning to surface the additional info we are adding, the access log just being a concrete place where it could happen.

Thanks for the pointer - I'll add support for logging this field to access_log_formatter.cc

Reply:
Removed the RC_ prefix